packages/Kokkos/test/tstCellList.cpp

```diff
@@ -0,0 +1,259 @@
+    auto fill_ghosts = KOKKOS_LAMBDA( const size_t i )
+    {
+        cell_list.is_ghost_cell( i ) = i % 2;
+    };
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, num_cells ),
+                          fill_ghosts );
+    Kokkos::fence();
+
+    // Calculate the number of ghosts on the device.
+    auto count_ghosts = KOKKOS_LAMBDA( const size_t i, unsigned &count )
+    {
+        count += cell_list.is_ghost_cell( i );
+    };
+    unsigned num_ghosts = 0;
+    Kokkos::parallel_reduce(
+        Kokkos::RangePolicy<ExecutionSpace>( 0, num_cells ), count_ghosts,
+        num_ghosts );
+
+    // Calculate the number of ghosts on the host.
+    auto host_ghosts = Kokkos::create_mirror_view( cell_list.is_ghost_cell );
+    Kokkos::deep_copy( host_ghosts, cell_list.is_ghost_cell );
+    unsigned host_num_ghost = 0;
+    for ( unsigned i = 0; i < num_cells; ++i )
+        host_num_ghost += host_ghosts( i );
+
+    // Check the number of ghosts.
+    TEST_EQUALITY( num_ghosts, host_num_ghost );
+    TEST_EQUALITY( num_ghosts, num_cells / 2 );
+
+    // Check the coordinates.
+    auto host_coords = Kokkos::create_mirror_view( cell_list.coordinates );
+    Kokkos::deep_copy( host_coords, cell_list.coordinates );
+    for ( unsigned i = 0; i < num_nodes; ++i )
+        for ( unsigned d = 0; d < space_dim; ++d )
+            TEST_EQUALITY( host_coords( i, d ), i + d );
+
+    // Check the cells.
+    auto host_cells = Kokkos::create_mirror_view( cell_list.cells );
+    Kokkos::deep_copy( host_cells, cell_list.cells );
+    for ( unsigned i = 0; i < num_cells; ++i )
+        for ( unsigned n = 0; n < nodes_per_cell; ++n )
+            TEST_EQUALITY( host_cells( i, n ), i + n );
+}
+
+//---------------------------------------------------------------------------//
+// Mixed topology test.
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( CellList, mixed_topology, NO )
+{
+    // Test types.
+    using DeviceType = typename NO::device_type;
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    // Create a list.
+    unsigned space_dim = 3;
+    unsigned num_cells = 1000;
+    unsigned num_nodes = 1200;
+    unsigned nodes_per_cell = 8;
+    size_t total_nodes_per_cell = num_cells * nodes_per_cell;
+    bool has_ghosts = true;
+    auto cell_list =
+        DataTransferKit::InputAllocators::createMixedTopologyCellList<
+            DeviceType>( space_dim, num_nodes, num_cells, total_nodes_per_cell,
+                         has_ghosts );
+
+    // Check sizes.
+    TEST_EQUALITY( Kokkos::rank( cell_list.coordinates ), 2 );
+    TEST_EQUALITY( cell_list.coordinates.extent( 0 ), num_nodes );
+    TEST_EQUALITY( cell_list.coordinates.extent( 1 ), space_dim );
+    TEST_EQUALITY( cell_list.cells.rank(), 1 );
+    TEST_EQUALITY( cell_list.cells.extent( 0 ), num_cells * nodes_per_cell );
+    TEST_EQUALITY( Kokkos::rank( cell_list.cell_topology_ids ), 1 );
+    TEST_EQUALITY( cell_list.cell_topology_ids.extent( 0 ), num_cells );
```

Comment:
Since this is a mixed topology test, you should check that it works with mixed topologies.